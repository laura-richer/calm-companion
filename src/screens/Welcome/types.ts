import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import { RootStackParamList } from '~components/Navigation/types';

export type WelcomeScreenNavigationProp = NativeStackNavigationProp<RootStackParamList, 'Welcome'>;

export type WelcomeProps = {
  navigation: WelcomeScreenNavigationProp;
};
