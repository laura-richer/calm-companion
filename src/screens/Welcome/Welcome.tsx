import React from 'react';
import { Button, Text, View, StyleSheet } from 'react-native';
import { WelcomeScreenNavigationProp } from './types';

const styles = StyleSheet.create({
  main: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    fontFamily: 'Poppins-Regular',
  },
  customText: {
    color: 'black',
    fontSize: 30,
  },
});

interface WelcomeProps {
  navigation: WelcomeScreenNavigationProp;
}

export const Welcome = ({ navigation }: WelcomeProps) => {
  return (
    <View style={styles.main}>
      <Text>Welcome</Text>
      <Button title="Get started" onPress={() => navigation.navigate('Home')} />
    </View>
  );
};
