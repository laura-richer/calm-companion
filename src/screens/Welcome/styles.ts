import { StyleSheet } from 'react-native';
import { fontFamily, colors, fontSize, spacer } from '~styles/variables';

export default StyleSheet.create({
  wrapper: {
    flex: 1,
    justifyContent: 'space-around',
    alignItems: 'center',
    paddingHorizontal: spacer.SM,
    paddingVertical: spacer.MD,
  },
  header: {
    gap: spacer.SM,
    alignItems: 'center',
  },
  footer: {
    gap: spacer.MD,
    alignItems: 'center',
    width: '100%',
  },
  title: {
    color: colors.WHITE,
  },
  subtitle: {
    fontFamily: fontFamily.MEDIUM,
    fontSize: fontSize.LG,
    color: colors.WHITE,
  },
  bodyText: {
    fontFamily: fontFamily.MEDIUM,
    fontSize: fontSize.MD,
    color: colors.WHITE,
    textAlign: 'center',
  },
  image: {
    maxWidth: '75%',
  },
});
