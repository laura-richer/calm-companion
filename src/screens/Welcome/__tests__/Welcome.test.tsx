import { screen, render } from '@testing-library/react-native';
import React from 'react';
import Welcome from '../Welcome';

describe('Welcome screen', () => {
  test('renders correct title', () => {
    render(<Welcome />);
    const button = screen.getByText('Get started');
    screen.debug();
    expect(button).toBeOnTheScreen();
  });
});
