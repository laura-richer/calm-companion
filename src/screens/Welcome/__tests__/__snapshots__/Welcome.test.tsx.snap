// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome screen matches snapshot 1`] = `
<BVLinearGradient
  angle={170}
  angleCenter={
    {
      "x": 0.75,
      "y": 0.75,
    }
  }
  colors={
    [
      4282350466,
      4287349689,
    ]
  }
  endPoint={
    {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "space-around",
      "paddingHorizontal": 20,
      "paddingVertical": 30,
    }
  }
  useAngle={true}
>
  <View
    style={
      {
        "alignItems": "center",
        "gap": 20,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#ffffff",
          },
          {
            "fontFamily": "Poppins-Bold",
            "fontSize": 34,
          },
        ]
      }
    >
      Calm Companion
    </Text>
    <Text
      style={
        {
          "color": "#ffffff",
          "fontFamily": "Poppins-Medium",
          "fontSize": 18,
        }
      }
    >
      Supporting your path to peace.
    </Text>
  </View>
  <Image
    resizeMode="contain"
    source={
      {
        "testUri": "../../../src/assets/images/calm-companion.png",
      }
    }
    style={
      {
        "maxWidth": "75%",
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "gap": 30,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#ffffff",
          "fontFamily": "Poppins-Medium",
          "fontSize": 14,
          "textAlign": "center",
        }
      }
    >
      Your handy reference of anxiety relief techniques and CBT tools to help you when your struggling.
    </Text>
    <View
      style={
        [
          {
            "shadowColor": "#8b8490",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 7,
          },
          {
            "alignItems": "center",
            "width": "100%",
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "backgroundColor": "#ffffff",
              "borderRadius": 50,
            },
            {
              "alignItems": "center",
              "width": "100%",
            },
          ]
        }
      >
        <Text
          style={
            {
              "color": "#3c433e",
              "fontFamily": "Poppins-SemiBold",
              "fontSize": 14,
              "padding": 10,
            }
          }
        >
          Get Started
        </Text>
      </View>
    </View>
  </View>
</BVLinearGradient>
`;
